libraries/NDP/src/NDP.cpp

```diff
@@ -239,24 +213,39 @@ int NDPClass::sendData(uint8_t *data, unsigned int len)
   return s;
 }
 
-
+int NDPClass::extractStart(void) {
+  uint32_t sample_size;
+  int s = syntiant_ndp120_tiny_get_recording_metadata(ndp,
+      &sample_size, SYNTIANT_NDP120_GET_FROM_MCU);
+  return s;
+}
 
 int NDPClass::extractData(uint8_t *data, unsigned int *len) {
   int s;
-  unsigned int l = audio_sample_chunk_size;
-  s = syntiant_ndp120_tiny_send_audio_extract(ndp, SYNTIANT_NDP120_EXTRACT_FROM_UNREAD);
+  unsigned int l = 768;
```

Comment:
if `sample_size` is made a member variable of the `NDP` class, then we don't have to hard code the 768 here, but use the `sample_size`instead